### PR TITLE
[MIG] report_py3o: Migration to 15.0

### PR DESCRIPTION
**Pull request by phuctranfxvn**
- Standard migration of module report_py3o to support odoo 15
- No extra feature added in this migration

### CODE REVIEW

**Comment by nilshamerlinck** [comment]
Nice! 

I believe the `ci/runbot` check could be disabled for this branch, right @gurneyalex ?

**Comment by BT-dmontull** [comment]
@nilshamerlinck @pedrobaeza @gurneyalex 
Does anyone know the status of this migration? it seems to be giving error in a check that it is no longer used.
thanks

**Review by gurneyalex — APPROVED** [review]

**Comment by gurneyalex** [comment]
@BT-dmontull I don't know much about the status of this migration apart from "it needs reviews". I'm not currently using this addon so I can't go beyond "the migration looks good to me". If you are asking, possibly this means you plan on using the module: could you then install it on a test instance and check that it works and make a review? This would allow me to merge the PR.

**Comment by gurneyalex** [comment]
@phuctranfxvn can you rebase your PR?

**Comment by StefanRijnhart** [comment]
@phuctranfxvn again, please rebase to resolve conflicts.

**Comment by phuctranfxvn** [comment]
> @phuctranfxvn again, please rebase to resolve conflicts.

Hello, sorry for late reply. conflicts resolved.

File: report_py3o/views/py3o_template.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        </field>
+    </record>
+    <record id="py3o_template_configuration_action" model="ir.actions.act_window">
+        <field name="name">Py3o Templates</field>
+        <field name="res_model">py3o.template</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+    <menuitem
+        id="py3o_template_configuration_menu"
+        parent="py3o_config_menu"
```
Comment (StefanRijnhart) [inline report_py3o/views/py3o_template.xml:50]: I don't see this template submenu in runboat (as user admin):

image


The same template menu does appear in the 14.0 runboat of this project:

image

Do you know why?

**Comment by FSchiepers** [comment]
Hello,

I think I found an error in the config description: In Odoo 15 it is no longer:

```
<?xml version="1.0" `encoding="utf-8"?>
<odoo>

<record id="account.account_invoices" model="ir.actions.report">
    <field name="report_type">py3o</field>
    <field name="py3o_filetype">odt</field>
    <field name="module">my_custom_module_base</field>
    <field name="py3o_template_fallback">report/account_invoice.odt</field>
</record>

</odoo>
```

but 

```
<?xml version="1.0" encoding="utf-8"?>
<odoo>

<record id="account_invoices" model="ir.actions.report">
    <field name="report_type">py3o</field>
    <field name="model">account.report</field>
    <field name="py3o_filetype">odt</field>
    <field name="module">my_custom_module_base</field>
    <field name="py3o_template_fallback">report/account_invoice.odt</field>
</record>

</odoo>
```

and you have to put 'account' in the manifest dependencies.

File: report_py3o/views/py3o_template.xml (hunk 1 above)
Comment (hoangtrann) [reply report_py3o/views/py3o_template.xml:50]: I think we should just drop this sub-menu entirely and make *Py3o Templates* be sub-menu of *Reporting*, which seems just fine?

**Comment by hoangtrann** [comment]
I've tested this PR and it works fine, although the js file naming with `esm` is a bit off from other sources

File: report_py3o/views/py3o_template.xml (hunk 1 above)
Comment (StefanRijnhart) [reply report_py3o/views/py3o_template.xml:50]: Sounds good!

**Comment by StefanRijnhart** [comment]
> I've tested this PR and it works fine, although the js file naming with `esm` is a bit off from other sources

That's according to the OCA guidelines, for ESLint compatibility.

Thanks for testing! Can you approve the PR on the `Files changed` tab?

File: report_py3o/views/py3o_template.xml (hunk 1 above)
Comment (hoangtrann) [inline report_py3o/views/py3o_template.xml:50]: I suggest dropping the sub-menu and we should be good to go
```suggestion
        parent="base.reporting_menuitem"
```

**Review by hoangtrann — APPROVED** [review]
Tested and work well on my end 👍 Thanks for the contribution

File: report_py3o/views/py3o_template.xml (hunk 1 above)
Comment (Tardo) [reply report_py3o/views/py3o_template.xml:50]: This change is mandatory to can see the Py3o models :/ I clarify that right now it is not possible to access the py3o models from the menu. Too many levels to be able to show itself.

**Comment by Tardo** [comment]
ping @phuctranfxvn

**Comment by Tardo** [comment]
I go to supersede this PR.

**Pull request closed without merging**